session-replacement/src/main/java/com/amadeus/session/servlet/SessionFilter.java

```diff
@@ -81,21 +82,45 @@ public void initForSession(FilterConfig config) {
   @Override
   public void doFilter(ServletRequest originalRequest, ServletResponse originalResponse, FilterChain chain)
       throws IOException, ServletException {
+
+    SessionManager sessionManager = (SessionManager)servletContext.getAttribute(Attributes.SESSION_MANAGER);
+    if (sessionManager == null) {
+      initSessionManagement(servletContext);
+    }
     ServletRequest request = prepareRequest(originalRequest, originalResponse, servletContext);
     ServletResponse response = prepareResponse(request, originalResponse, servletContext);
-    SessionManager sessionManager = (SessionManager)servletContext.getAttribute(Attributes.SESSION_MANAGER);
 
     try {
+
+      logger.debug("Start of SessionFilter doFilter ");
       // Call next filter in chain
       chain.doFilter(request, response);
+
+      logger.debug("after the doFilter of SessionFilter doFilter ");
+
+    } catch (Exception e) {
+
+      logger.error("Error in sessionFiler catch", e);
```

Comment:
typo "Session filter" + maybe we should not log stack trace here as the exception is propagated.

Reply:
typo fixed 
i prefer to keep the logging